src/main/kotlin/org/rust/cargo/toolchain/RustToolchain.kt

```diff
@@ -176,6 +188,23 @@ private fun parseRustcVersion(lines: List<String>): Version? {
         UnstableVersion(commitHash, major, minor, build)
 }
 
+private fun parseRustupVersion(lines: List<String>): Version? {
+    // We want to parse following
+    //
+    //  ```
+    //  rustup 0.5.0 (4be1012 2016-07-30)
+    //  ```
+    val releaseRe = """rustup (\d+)\.(\d+)\.(\d+) \(.*\)""".toRegex()
+
+    val match = lines.mapNotNull { releaseRe.matchEntire(it) }.firstOrNull() ?: return null
+
+    val major = match.groups[1]!!.value.toInt()
+    val minor = match.groups[2]!!.value.toInt()
+    val build = match.groups[3]!!.value.toInt()
```

Comment:
if the output doesn't match regexp, it will return null for `Version?` and show `N/A` on the panel just like `rustc` and `cargo`.